server/services/applicationService.ts

```diff
@@ -59,10 +60,24 @@ export default class ApplicationService {
       const pageName = getPageName(page.constructor)
       const taskName = getTaskName(page.constructor)
 
+      const oldBody = application.data?.[taskName]?.[pageName]
+
       application.data = application.data || {}
       application.data[taskName] = application.data[taskName] || {}
       application.data[taskName][pageName] = page.body
 
+      const checkYourAnswersTaskName = getTaskName(CheckYourAnswers)
+      const checkYourAnswersPageName = getPageName(CheckYourAnswers)
+
+      if (
+        application.data[checkYourAnswersTaskName] &&
+        !(taskName === checkYourAnswersTaskName && pageName === checkYourAnswersPageName)
+      ) {
+        if (!oldBody || !pageBodyShallowEquals(oldBody, page.body)) {
```

Comment:
would it be helpful to wrap this rather opaque test in a local function with an intention-revealing name e.g. `anotherSectionHasSinceBeenEdited()` (if that's what this actually does?!)?

Reply:
I definitely intended to pull this out into its own function, so I agree an a refactor would be useful here 👍🏾

Reply:
I've done the renaming too... if I understand right it's checking whether a specific page has changed, so named accordingly